### ISSUE TITLE
can't install with Python3

### ISSUE THREAD

**Issue by Horace89** [spotify/annoy#90]
Hi. Newbie to Github here. Thanks for this wonderful piece of software.

I think I've a small bug in the install with Python 3:
```
(annoy)horace89@horace89:~/tst1/annoy$ python -V
Python 3.4.0
(annoy)horace89@horace89:~/tst1/annoy$ pip install annoy              
Collecting annoy
  Using cached annoy-1.4.1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/tmp/pip-build-ygnacgga/annoy/setup.py", line 34, in <module>
        long_description = readme_note + fobj.read()
      File "/home/horace89/.virtualenvs/annoy/lib/python3.4/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 3767: ordinal not in range(128)
```

If I build a virtualenv with Python2.7, the error disappears and the build is ok.
If I use a virtualenv with Python3.4 and if I edit the setup.py file with (I comment out fobj.read()) :
```with open('README.rst') as fobj:
long_description = readme_note #+ fobj.read()```
, the error disappears and the build is ok.
So I think, there's an issue with the utf-8 encoding.

**Comment by erikbern** [comment]
Interesting – I think probably just wrapping the file reading in a try/except should resolve the issue – I can do later today

**Comment by Horace89** [comment]
No problem. I just wanted to inform you.

**Comment by a1k0n** [comment]
I think you need to use `with open('README.rst', encoding='utf-8') as fobj:` -- the default encoding for `open` is `ascii`. Don't know why Python 3.x blows up and 2.x doesn't...

**Comment by erikbern** [comment]
Travis is building it on Python 3 so I don't get it. Are you on Win or something?

**Comment by Horace89** [comment]
I'm on Ubuntu 14.04.

**Comment by erikbern** [comment]
weird will try to repro

**Comment by erikbern** [comment]
Can't repro on Ubuntu 14.04. Anything else that could be special about your setup.

I could fix the bug though, just nicer to be able to repro it

```
$ python3.4 setup.py build
running build
running build_py
running build_ext
building 'annoy.annoylib' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.4m -c src/annoymodule.cc -o build/temp.linux-x86_64-3.4/src/annoymodule.o -O3 -march=native -ffast-math
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.4/src/annoymodule.o -o build/lib.linux-x86_64-3.4/annoy/annoylib.cpython-34m.so
```

**Comment by Horace89** [comment]
Really really weird... 
I just tried at home on my *local* install of Ubuntu 14.04 and I'm not able to reproduce the bug.
I found the previous issue when using the default (Ubuntu 14.04) VM on koding.com. I guess it comes from their config, but I don't know what is special to it.

**Comment by erikbern** [comment]
My PR didn't work either. Can you figure out a way to fix it and let me know? Possibly send a PR if you want to

**Comment by Horace89** [comment]
First I must learn *how* to do pull requests :wink:

**Comment by a1k0n** [comment]
@erikbern: to fix your PR in a python 2.x-compatible way you have to `import codecs`. `codecs.open('README.rst', encoding='utf-8')` should work then.

**Comment by Horace89** [comment]
ok. On my koding.com VM, I edited setup.py. I just changed
```with open('README.rst') as fobj:```
to
```with open('README.rst', encoding='utf-8') as fobj:```
and now the the install finishes flawlessly. Thanks to @a1k0n for his comment!

**Comment by erikbern** [comment]
I fixed it using codecs now in #91

Will push new version to PyPI
